sbi/inference/posteriors/direct_posterior.py

```diff
@@ -582,6 +573,32 @@ def __call__(
         else:
             NotImplementedError
 
+    def get_potential(self, theta: Tensor, track_gradients: bool = False) -> Tensor:
```

Comment:
This is quite a big refactor. Please run slow tests to make sure that everything runs (e.g. rejection sampling)

Reply:
I think the refactor makes a lot of general. We could also have **only** `posterior_potential` and `pyro_potential`. Then, `pyro_potential` does the same as it does in your implementation but then calls `posterior_potential` instead of `get_potential`. No?

Reply:
yes, makes sense, will do